.circleci/config.yml

```diff
@@ -10,42 +10,37 @@ aliases:
     name: setup_miniconda
     command: |
       mkdir -p workspace
-      git clone -b validateNightly git@github.com:CDAT/cdat workspace/cdat
+      git clone -b validateNightly --depth 1 git@github.com:CDAT/cdat workspace/cdat
       ls workspace/cdat
-      # following will install miniconda3 under $WORKDIR/miniconda/bin
       python workspace/cdat/scripts/install_miniconda.py -w $WORKDIR -p 'py3'
       
   - &create_conda_env
     name: create_conda_env
+    environment:
+       CHANNELS: "-c cdat/label/unstable -c cdat/label/nightly -c conda-forge -c cdat"
+       PKGS: "libcf distarray cdtime libcdms cdat_info testsrunner numpy esmf esmpy libdrs_f pyopenssl nose requests flake8 myproxyclient"
     command: |
        export PATH=$HOME/project/$WORKDIR/miniconda/bin:$PATH
        conda config --set always_yes yes --set changeps1 no
        conda update -y -q conda
        conda config --set anaconda_upload no
-       conda create -n py3 -c cdat/label/unstable -c cdat/label/nightly -c conda-forge -c cdat libcf distarray cdtime libcdms cdat_info testsrunner numpy esmf esmpy libdrs_f pyopenssl nose requests flake8 myproxyclient "python>3"
-       conda create -n py2 -c cdat/label/unstable -c cdat/label/nightly -c conda-forge -c cdat libcf distarray cdtime libcdms cdat_info testsrunner numpy esmf esmpy libdrs_f pyopenssl nose requests flake8 myproxyclient "python<3"
+       if [[ $PY_VER = 'py2' ]]; then
+          conda create -n $PY_VER $CHANNELS $PKGS "python<3"
+       else
+          conda create -n $PY_VER $CHANNELS $PKGS "python>3" coverage coveralls
+       fi
        if [ $(uname) == "Linux" ]; then
-           conda install -n py3 -c cdat/label/unstable -c cdat/label/nightly -c conda-forge gcc_linux-64
-           conda install -n py2 -c cdat/label/unstable -c cdat/label/nightly -c conda-forge gcc_linux-64
+           conda install -n $PY_VER -c cdat/label/unstable -c cdat/label/nightly -c conda-forge gcc_linux-64
        else
-           conda install -n py3 -c cdat/label/unstable -c cdat/label/nightly -c conda-forge gcc
-           conda install -n py2 -c cdat/label/unstable -c cdat/label/nightly -c conda-forge gcc
+           conda install -n $PY_VER -c cdat/label/unstable -c cdat/label/nightly -c conda-forge gcc
```

Comment:
if you look at cmor config file I have an env variable that decides the C_COMPILER to use for Linux or Mac.